### DISCUSSION TITLE
__init__ in Renderer

### DISCUSSION THREAD

**Discussion by PinkMushroom** [Gnimuc/CImGui.jl#57] [Q&A] [answered]
Hi,

In the Renderer class there is a __init__ function and i can't figure out how it gets invoked.
(I thought for sure it was a magic function that gets invoked once for a module, i.e. a Julia thing, but i can find no reference to such a thing in the Julia documentation).

Here's the code:
```
function __init__()
    println("__init__")
    @static if Sys.isapple()
        # OpenGL 3.2 + GLSL 150
        global glsl_version = 150
        GLFW.WindowHint(GLFW.CONTEXT_VERSION_MAJOR, 3)
        GLFW.WindowHint(GLFW.CONTEXT_VERSION_MINOR, 2)
        GLFW.WindowHint(GLFW.OPENGL_PROFILE, GLFW.OPENGL_CORE_PROFILE) # 3.2+ only
        GLFW.WindowHint(GLFW.OPENGL_FORWARD_COMPAT, GL_TRUE) # required on Mac
    else
        # OpenGL 3.0 + GLSL 130
        global glsl_version = 130
        GLFW.WindowHint(GLFW.CONTEXT_VERSION_MAJOR, 3)
        GLFW.WindowHint(GLFW.CONTEXT_VERSION_MINOR, 0)
        # GLFW.WindowHint(GLFW.OPENGL_PROFILE, GLFW.OPENGL_CORE_PROFILE) # 3.2+ only
        # GLFW.WindowHint(GLFW.OPENGL_FORWARD_COMPAT, GL_TRUE) # 3.0+ only
    end
end
```

**Comment by Gnimuc** [comment, marked as answer]
> I thought for sure it was a magic function that gets invoked once for a module

Yes, you're right, and you can read more about the use case here: https://docs.julialang.org/en/v1/manual/modules/#Module-initialization-and-precompilation

**Reply by PinkMushroom** [reply]
thank you! if you enter  `__init__ ` in the search box in Julia docs it returns nothing...
